### PR TITLE
[2100] Reconcile a namespace again with missing rolebinding

### PR DESCRIPTION
**Pull request by jkhelil**
# Changes
fixess https://github.com/tektoncd/operator/issues/2100

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes
None

### CODE REVIEW

File: pkg/reconciler/openshift/tektonconfig/rbac.go [hunk 1]
```diff
@@ -261,6 +261,13 @@ func (r *rbac) getNamespacesToBeReconciled(ctx context.Context) ([]corev1.Namesp
...
 		sccRoleBinding, err := r.kubeClientSet.RbacV1().RoleBindings(ns.Name).Get(ctx, pipelinesSCCRoleBinding, metav1.GetOptions{})
 		if err != nil {
+			// https://github.com/tektoncd/operator/issues/2100
+			// Reconcile a namespace again with missing RoleBinding
+			if errors.IsNotFound(err) {
+				logger.Infof("RoleBinding %s is missing in namespace: %s, will reconcile again the namespace %s", pipelinesSCCRoleBinding, ns.Name, ns.Name)
+				namespaces = append(namespaces, ns)
+				continue
+			}
 			return nil, err
```
Comment (piyush-garg) [inline pkg/reconciler/openshift/tektonconfig/rbac.go:271]: can we custumize this error a bit also with error fetching rolebinding `name` from `namespace`: err

File: pkg/reconciler/openshift/tektonconfig/rbac.go [hunk 2]
```diff
@@ -261,6 +261,13 @@ func (r *rbac) getNamespacesToBeReconciled(ctx context.Context) ([]corev1.Namesp
 		// We must make sure that the default SCC is in force via the ClusterRole.
 		sccRoleBinding, err := r.kubeClientSet.RbacV1().RoleBindings(ns.Name).Get(ctx, pipelinesSCCRoleBinding, metav1.GetOptions{})
 		if err != nil {
+			// https://github.com/tektoncd/operator/issues/2100
+			// Reconcile a namespace again with missing RoleBinding
+			if errors.IsNotFound(err) {
+				logger.Infof("RoleBinding %s is missing in namespace: %s, will reconcile again the namespace %s", pipelinesSCCRoleBinding, ns.Name, ns.Name)
```
Comment (piyush-garg) [inline pkg/reconciler/openshift/tektonconfig/rbac.go:267]: SHould we put this into debug

**Review by concaf — APPROVED** [review]
this will make a lot of our headaches go away 🎉 thanks 🙂

**Comment by concaf** [comment]
we should also cherry pick this to older releases

**Review by jkandasa — APPROVED** [review]

File: pkg/reconciler/openshift/tektonconfig/rbac.go (hunk 1 above)
Comment (jkhelil) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:271]: yes done thanks @piyush-garg

File: pkg/reconciler/openshift/tektonconfig/rbac.go (hunk 2 above)
Comment (jkhelil) [reply pkg/reconciler/openshift/tektonconfig/rbac.go]: done

File: pkg/reconciler/openshift/tektonconfig/rbac.go (hunk 1 above)
Comment (piyush-garg) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:271]: I think I was not clear, error message in the logger was fine, i am saying about this for the error we are returning to the end

Comment (jkandasa) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:271]: If I understand correctly, you asking to print the error message with the logger on line number 271, then return `nil, err`.
Am I correct @piyush-garg ?

Comment (jkhelil) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:271]: I think @piyush-garg is asking to wrap the error

File: pkg/reconciler/openshift/tektonconfig/rbac.go [hunk 3]
```diff
@@ -261,7 +261,14 @@ func (r *rbac) getNamespacesToBeReconciled(ctx context.Context) ([]corev1.Namesp
 		// We must make sure that the default SCC is in force via the ClusterRole.
 		sccRoleBinding, err := r.kubeClientSet.RbacV1().RoleBindings(ns.Name).Get(ctx, pipelinesSCCRoleBinding, metav1.GetOptions{})
 		if err != nil {
-			return nil, err
+			// https://github.com/tektoncd/operator/issues/2100
```
Comment (jkandasa) [inline pkg/reconciler/openshift/tektonconfig/rbac.go:264]: @jkhelil we may remove this GH issue reference
(AFAIK, we do not have incomplete work from this issue mentioned here.)

File: pkg/reconciler/openshift/tektonconfig/rbac.go (hunk 1 above)
Comment (piyush-garg) [reply pkg/reconciler/openshift/tektonconfig/rbac.go]: yes to wrap the error and return, not logger

**Pull request merged**